packages/protocol/engine/clock/blocktime/clock.go

```diff
@@ -0,0 +1,75 @@
+package blocktime
+
+import (
+	"github.com/iotaledger/goshimmer/packages/core/module"
+	"github.com/iotaledger/goshimmer/packages/protocol/engine"
+	"github.com/iotaledger/goshimmer/packages/protocol/engine/clock"
+	"github.com/iotaledger/goshimmer/packages/protocol/engine/consensus/blockgadget"
+	"github.com/iotaledger/hive.go/core/slot"
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/runtime/event"
+	"github.com/iotaledger/hive.go/runtime/options"
+)
+
+// Clock implements the clock.Clock interface that sources its notion of time from accepted and confirmed blocks.
+type Clock struct {
+	// acceptedTime contains a notion of time that is anchored to the latest accepted block.
+	acceptedTime *RelativeTime
+
+	// confirmedTime contains a notion of time that is anchored to the latest confirmed block.
+	confirmedTime *RelativeTime
+
+	// Module embeds the required methods of the module.Interface.
+	module.Module
+}
+
+// NewProvider creates a new Clock provider with the given options.
+func NewProvider(opts ...options.Option[Clock]) module.Provider[*engine.Engine, clock.Clock] {
+	return module.Provide(func(e *engine.Engine) clock.Clock {
+		return options.Apply(&Clock{
+			acceptedTime:  NewRelativeTime(),
+			confirmedTime: NewRelativeTime(),
+		}, opts, func(c *Clock) {
+			e.HookConstructed(func() {
+				e.LedgerState.HookInitialized(func() {
+					c.acceptedTime.Set(e.SlotTimeProvider().EndTime(e.Storage.Settings.LatestCommitment().Index()))
+					c.confirmedTime.Set(e.SlotTimeProvider().EndTime(e.Storage.Settings.LatestCommitment().Index()))
+
+					c.TriggerInitialized()
+				})
+
+				e.Events.Clock.AcceptedTimeUpdated.LinkTo(c.acceptedTime.OnUpdated)
+				e.Events.Clock.ConfirmedTimeUpdated.LinkTo(c.confirmedTime.OnUpdated)
+
+				async := event.WithWorkerPool(e.Workers.CreatePool("Clock", 1))
```

Comment:
don't call the workerpool `async`, this makes it confusing

Reply:
It's not the workerpool that is called async but the "option" to "execute things asyncronously in a workerpool"

I personally think it makes the code more readable to name the variable in such a qualitative way rather than "workerPoolOption" or sth. which would be more aligned with the low-level concepts behind the async hooking

Reply:
The problem I have with `async` is that this name is a qualifier in lots of languages. A variable name should imho explain what it contains. The only variables where I would really name them more explaining the behaviour than the content are boolean flags.